vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/client/MobitruFacadeImpl.java

```diff
@@ -85,18 +92,35 @@ public String takeDevice(DesiredCapabilities desiredCapabilities) throws Mobitru
         Object deviceUdid = capabilities.getOrDefault(APPIUM_UDID, capabilities.get(UDID));
         if (deviceUdid != null)
         {
-            //use different API in case if udid is provided in capabilities
-            //it's required in some cases like if the device is already taken
-            LOGGER.info("Trying to take device with udid {}", deviceUdid);
-            return takeDevice(getDefaultDeviceWaiter(),
-                    () -> mobitruClient.takeDeviceBySerial(String.valueOf(deviceUdid)),
-                    () -> "Unable to take device with udid " + deviceUdid);
+            return takeDevice(Set.of(String.valueOf(deviceUdid)));
         }
         Device device = new Device();
         device.setDesiredCapabilities(desiredCapabilities.asMap());
         return takeDevice(device, getDefaultDeviceWaiter());
     }
 
+    private String takeDevice(Set<String> udids) throws MobitruOperationException
+    {
+        String udidsAsString = String.join(", ", udids);
+        final String errorMessage = "Unable to take device with udid " + udidsAsString;
+        return takeDevice(getDefaultDeviceWaiter(), () ->
+        {
+            for (String udid : udids)
+            {
+                LOGGER.atInfo().addArgument(udid).log("Trying to take device with udid {}");
+                try
+                {
+                    return mobitruClient.takeDeviceBySerial(udid);
+                }
+                catch (MobitruOperationException e)
+                {
+                    LOGGER.atWarn().setCause(e).addArgument(udid).log("Unable to take device with udid {}");
```

Comment:
if `udids.size() == 1`, this message doesn't make sense, as exception is thrown immediately